Unity/opentok/holo/holo/Capturer.mm

```diff
@@ -0,0 +1,373 @@
+    return Holographic::Compression::compress(inputArray, inputSize, outputArray, outputSize);
+}
+
+- (BOOL)compressInputArray:(nonnull CVPixelBufferRef)depthDataMap outputArray:(std::unique_ptr<uint8_t[]> &)outputArray outputSize:(uint32_t &)outputSize {
+    return Holographic::Compression::compress(depthDataMap, outputArray, outputSize);
+}
+
+@end
+
+@interface Capturer()
+@property (nonatomic) RTC_OBJC_TYPE(VonageRTCCameraVideoCapturer) *capturer;
+@property (nonatomic) DepthDataCompressor *depthDataCompressor;
+
+- (void) setupListenerBlocks;
+@end
+
+@implementation Capturer {
+    __weak id<OTVideoCaptureConsumer> _videoCaptureConsumer;
+    OTVideoFrame* _videoFrame;
+
+    uint32_t _captureWidth;
+    uint32_t _captureHeight;
+    NSString* _capturePreset;
+
+    BOOL _capturing;
+
+    enum OTCapturerErrorCode _captureErrorCode;
+}
+
+@synthesize videoCaptureConsumer = _videoCaptureConsumer;
+@synthesize videoContentHint;
+
+-(id)init {
+    self = [super init];
+    if (self) {
+        _capturePreset = AVCaptureSessionPreset640x480;
+        [[self class] dimensionsForCapturePreset:_capturePreset
+                                           width:&_captureWidth
+                                          height:&_captureHeight];
+        _capture_queue = dispatch_queue_create("com.vonage.holo.Capturer",
+                                               DISPATCH_QUEUE_SERIAL);
+        _videoFrame = [[OTVideoFrame alloc] initWithFormat:
+                       [OTVideoFormat videoFormatNV12WithWidth:_captureWidth
+                                                        height:_captureHeight]];
+    }
+    return self;
+}
+
+-(id)initWithCapturePreset: (NSString*)preset {
+    self = [super init];
+    if (self) {
+        _capturePreset = preset;
+        [[self class] dimensionsForCapturePreset:_capturePreset
+                                           width:&_captureWidth
+                                          height:&_captureHeight];
+        _capture_queue = dispatch_queue_create("com.vonage.holo.Capturer",
+                                               DISPATCH_QUEUE_SERIAL);
+        _videoFrame = [[OTVideoFrame alloc] initWithFormat:
+                       [OTVideoFormat videoFormatNV12WithWidth:_captureWidth
+                                                        height:_captureHeight]];
+    }
+    return self;
+}
+
+- (void)dealloc {
+    [self stopCapture];
+    [self releaseCapture];
+
+    if (_capture_queue) {
+        _capture_queue = nil;
+    }
+    _videoFrame = nil;
+}
+
+- (void)initCapture {
+    _capturer = [[RTC_OBJC_TYPE(VonageRTCCameraVideoCapturer) alloc] init];
+    _depthDataCompressor = [[DepthDataCompressor alloc] init];
+
+    [_capturer setDelegate:self];
+    [_capturer updateDepthDelegate:_depthDataCompressor];
+}
+
+- (void)releaseCapture {
+    [self stopCapture];
+}
+
+- (int32_t)startCapture {
+    _capturing = YES;
+    dispatch_after(dispatch_time(DISPATCH_TIME_NOW, 1 * NSEC_PER_SEC),
```

Comment:
why do we need to wait 1 second?
if there is no reason should we make the call `dispatch_sync` or `dispatch_async`?

Reply:
i don't see a reason for 1 sec timeout... but lets add it as it is, i will fix it after.